arches/app/models/resource.py

```diff
@@ -146,36 +148,36 @@ def get_descriptor(self, descriptor, context):
             except KeyError:
                 pass
 
-        self.calculate_descriptors(context=context)
-        return self.descriptors[language][descriptor]
-
-    def calculate_descriptors(self, descriptors=("name", "description", "map_popup"), context=None):
+    def save_descriptors(self, descriptors=("name", "description", "map_popup"), context=None):
         """
         descriptors -- iterator with descriptors to be calculated
-        context -- Dictionary which may have:
-            language -- Language code in which the descriptor should be returned (e.g. 'en').
-                This occurs when handling concept values.
-            any key:value pairs needed to control the behavior of a custom descriptor function
+        context -- Dictionary with any key:value pairs needed to control the behavior of a custom descriptor function
 
         """
 
-        language = self.get_descriptor_language(context)
-
-        if not self.descriptor_function:
-            self.descriptor_function = models.FunctionXGraph.objects.filter(
-                graph_id=self.graph_id, function__functiontype="primarydescriptors"
-            ).select_related("function")
-
-        for descriptor in descriptors:
-            if len(self.descriptor_function) == 1:
-                module = self.descriptor_function[0].function.get_class_module()()
-                self.descriptors[language][descriptor] = module.get_primary_descriptor_from_nodes(
-                    self, self.descriptor_function[0].config["descriptor_types"][descriptor], context
-                )
-                if descriptor == "name" and self.descriptors[language][descriptor] is not None:
-                    self.name[language] = self.descriptors[language][descriptor]
+        for lang in settings.LANGUAGES:
```

Comment:
I think we should only attempt to save the descriptor if the `context["tile"].sortorder` > 0